lib/libc/string/wcscasecmp.c

```diff
@@ -41,7 +42,9 @@ wcscasecmp(const wchar_t *s1, const wchar_t *s2)
 		c1 = towlower(*s1);
 		c2 = towlower(*s2);
 		if (c1 != c2)
-			return ((int)c1 - c2);
+			return ((uintmax_t)c1 > (uintmax_t)c2 ? 1 : -1);
```

Comment:
so why is the cast needed here? they are the same time, why can't the comparison be made without one?
This is odd-ball enough that a comment about the relatively obscure corner of 'C' that necessitates this would be good too.

Reply:
Answer: w_char is not guaranteed to be an integer, but yeah I think this could be done better and it will.